backend/apis/leads/update-lead.api.js

```diff
@@ -83,6 +83,47 @@ app.patch("/api/leads/:id", (req, res, next) => {
       );
     }
 
+    const leadServicesChanged = atLeastOne(req.body, "leadServices");
+
+    if (leadServicesChanged && fullLead.leadServices.length > 0) {
+      queries.push("UPDATE leads SET modifiedBy = ? WHERE id = ?;");
+      queryData.push(userId, leadId);
+
+      const oldLeadServices = [...oldLead.leadServices];
```

Comment:
There doesn't seem to be a purpose for this array existing. We can just delete this line and then change line 93 to be `oldLead.leadServices.map`